packages/smooth_app/lib/temp/product_preferences.dart

```diff
@@ -41,151 +87,131 @@ class UserPreferencesModel extends ChangeNotifier {
       attributeGroupString =
-          await assetBundle.loadString(_getAttributeAssetPath(languageCode));
-    } catch (e) {
-      /// we don't have all the languages in the assets
-      return null;
-    }
-    try {
-      final UserPreferencesModel userPreferencesModel =
-          UserPreferencesModel._();
-      userPreferencesModel._languageCode = languageCode;
-      userPreferencesModel._isHttps = false;
-      if (userPreferencesModel._loadStrings(
-          importanceString, attributeGroupString)) {
-        return userPreferencesModel;
-      }
+          await assetBundle.loadString(attributeGroupAssetPath);
+      return loadReferenceFromStrings(
+        importanceString,
+        attributeGroupString,
+        languageCode,
+        false,
+      );
     } catch (e) {
       print('An error occurred while loading user preferences : $e');
     }
-    return null;
+    return false;
   }
 
-  static Future<UserPreferencesModel> _getHttps(
-    final String languageCode,
-  ) async {
+  /// Loads the references of importance and attribute groups from urls.
+  Future<bool> loadReferenceFromHttps(
+    final String languageCode, {
+    String importanceUrl,
+    String attributeGroupUrl,
+  }) async {
     try {
-      final String importanceUrl = PreferenceImportance.getUrl(languageCode);
-      final String attributeUrl = _getAttributeUrl(languageCode);
+      importanceUrl ??= PreferenceImportance.getUrl(languageCode);
+      attributeGroupUrl ??= _getAttributeUrl(languageCode);
       http.Response response;
       response = await http.get(Uri.parse(importanceUrl));
-      // TODO(monsieurtanuki): check response.statusCode
+      if (response.statusCode != 200) {
+        return false;
+      }
       final String importanceString = response.body;
-      response = await http.get(Uri.parse(attributeUrl));
-      // TODO(monsieurtanuki): check response.statusCode
-      final String attributeGroupString = response.body;
-      final UserPreferencesModel userPreferencesModel =
-          UserPreferencesModel._();
-      userPreferencesModel._languageCode = languageCode;
-      userPreferencesModel._isHttps = true;
-      if (userPreferencesModel._loadStrings(
-          importanceString, attributeGroupString)) {
-        return userPreferencesModel;
+      response = await http.get(Uri.parse(attributeGroupUrl));
+      if (response.statusCode != 200) {
+        return false;
       }
+      final String attributeGroupString = response.body;
+      return loadReferenceFromStrings(
+        importanceString,
+        attributeGroupString,
+        languageCode,
+        true,
+      );
     } catch (e) {
       print('An error occurred while loading user preferences : $e');
     }
-    return null;
+    return false;
   }
 
-  bool _loadJsonString(final String inputString, final Function fromJson) {
+  /// Loads the references of importance and attribute groups from json strings.
+  bool loadReferenceFromStrings(
```

Comment:
can we reanme it to loadReferenceFromJSONStrings? It puzzled me a bit at first sight.